src/type_sys/fender_reference/mod.rs

```diff
@@ -134,7 +134,7 @@ impl Value for FenderReference {
 
     fn into_ref(self) -> Self {
         match self {
-            FenderReference::FRef(_) => self,
+            FenderReference::FRef(_) => FenderReference::FRef(self.deref().clone().into()),
```

Comment:
I'm confused here 
is this makeing a copy of the value in the reference or is it still pointing to the same thing?
we want this to return the existing reference if it already is one